src/metabase/legacy_mbql/schema.cljc

```diff
@@ -545,7 +545,7 @@
   s StringExpressionArg, start IntGreaterThanZeroOrNumericExpression, length (optional NumericExpressionArg))
 
 (defclause ^{:requires-features #{:expressions :split-part}} split-part
-  text StringExpressionArg, delimiter ExpressionArg, position IntGreaterThanZeroOrNumericExpression)
+  text StringExpressionArg, delimiter [:string {:min 1}], position IntGreaterThanZeroOrNumericExpression)
```

Comment:
Can you use `::lib.schema.common/non-blank-string` here? I'm not sure what the rule is, and both appear in various places, but `non-blank-string` seems far more common and possibly provides a nicer error message.